daemon/protocol/python/tests/run_many.py

```diff
@@ -0,0 +1,200 @@
+
+        proto.start()
+
+        #wait up to 1 second for daemon to start 
+        start = time.time()
+        elapsedTime = 0
+        started = False
+
+        while not started and elapsedTime < 1.0:    
+            try:
+                echo = proto.echo("ping")
+                if echo == "ping":
+                    started = True
+            except proto.ScootException as e:
+                elapsedTime = time.time() - start
+        
+        if not started:
+            self.fail("Daemon didn't start within {0}".format(elapsedTime))
+                        
+    def tearDown(self):
+        self.daemonProcess.kill()
+        unittest.TestCase.tearDown(self)
+        
+    sleep_ss_key = "sleep_s_id"
+    ls_ss_key = "ls_s_id"
+    ls_fail_ss_key = "ls_fail_s_id"
+    
+    # test scenario
+    class CmdDef(object):
+        def __init__(self, cmd, snapshot_key, interm_states, final_state):
+            self.cmd = cmd
+            self.snapshot_key = snapshot_key
+            self.interm_states = interm_states
+            self.final_state = final_state
+        
+    sleepDef = CmdDef(cmd=["./sleep_script.sh"], 
+                      snapshot_key="sleep_s_id", 
+                      interm_states=[proto.ScootStatus.State.PENDING, 
+                                     proto.ScootStatus.State.PREPARING,
+                                     proto.ScootStatus.State.RUNNING, 
+                                     proto.ScootStatus.State.COMPLETED], 
+                      final_state=proto.ScootStatus.State.COMPLETED)  # run sleep 1, expect complete or running, final status:complete
+    lsDef = CmdDef(cmd=["./ls_script.sh"], 
+                   snapshot_key="ls_s_id", 
+                   interm_states=[proto.ScootStatus.State.PREPARING, 
+                                  proto.ScootStatus.State.PENDING, 
+                                  proto.ScootStatus.State.RUNNING, 
+                                  proto.ScootStatus.State.COMPLETED], 
+                   final_state=proto.ScootStatus.State.COMPLETED)  # run ls script 5 times expect complete, running or queued, final status:complete
+    failDef = CmdDef(cmd=["./ls_script.sh"], 
+                     snapshot_key="ls_fail_s_id", 
+                     interm_states=[proto.ScootStatus.State.PENDING,
+                                    proto.ScootStatus.State.PREPARING, 
+                                    proto.ScootStatus.State.RUNNING, 
+                                    proto.ScootStatus.State.COMPLETED, 
+                                    proto.ScootStatus.State.FAILED], 
+                     final_state=proto.ScootStatus.State.COMPLETED)  # run ls script 5 times expect complete, running or queued, final status:failed
+
+    # TODO start the server    
+    def test_many(self):
+        tmpdir = tempfile.mkdtemp()
+        rpc_timeout_ns = int(1000 * 1e6)
+        
+        runs = {}
+        
+        try:
```

Comment:
Same comment about this try block, it seems unnecessary, we could make the tmpdir in setup and remove it in teardown.  Also, it looks like this tmpdir isn't used, and then we make a second one in make_snapshots which doesn't get cleaned up?